cloudera/tests/conftest.py

```diff
@@ -49,47 +53,154 @@ def config():
+    return deepcopy(common.INSTANCE)
+
+
+@pytest.fixture
+def instance_bad_url():
+    return deepcopy(common.INSTANCE_BAD_URL)
+
+
+@pytest.fixture
+def instance_autodiscover_include_not_array():
+    return deepcopy(common.INSTANCE_AUTODISCOVER_INCLUDE_NOT_ARRAY)
+
+
+@pytest.fixture
+def instance_autodiscover_include_with_one_entry_dict():
+    return deepcopy(common.INSTANCE_AUTODISCOVER_INCLUDE_WITH_ONE_ENTRY_DICT)
+
+
+@pytest.fixture
+def instance_autodiscover_include_with_two_entries_dict():
+    return deepcopy(common.INSTANCE_AUTODISCOVER_INCLUDE_WITH_TWO_ENTRIES_DICT)
+
+
+@pytest.fixture
+def instance_autodiscover_include_with_str():
+    return deepcopy(common.INSTANCE_AUTODISCOVER_INCLUDE_WITH_STR)
+
+
+@pytest.fixture
+def instance_autodiscover_exclude():
+    return deepcopy(common.INSTANCE_AUTODISCOVER_EXCLUDE)
 
 
 @pytest.fixture
 def init_config():
-    return common.INIT_CONFIG
+    return deepcopy(common.INIT_CONFIG)
 
 
 @pytest.fixture(scope='session')
 def cloudera_check():
     return lambda instance: deepcopy(ClouderaCheck('cloudera', init_config=common.INIT_CONFIG, instances=[instance]))
 
 
+def get_timeseries_resource(query):
+    return ApiTimeSeriesResponseList(
+        items=[
+            ApiTimeSeriesResponse(
+                time_series=[
+                    ApiTimeSeries(
+                        data=[
+                            ApiTimeSeriesData(value=49.7),
+                        ],
+                        metadata=ApiTimeSeriesMetadata(attributes={'category': category}, alias=metric),
+                    )
+                    for metric in metrics
+                ]
+            )
+            for category, metrics in TIMESERIES_METRICS.items()
+            if re.search(f'category={category.upper()}', query)
+        ]
+    )
+
+
 @pytest.fixture
-def api_response():
-    def _response(filename):
-        with open(os.path.join(common.HERE, "api_responses", f'{filename}.json'), 'r') as f:
-            return json.load(f)
+def cloudera_api_exception(request):
+    return ApiException(request.param)
+
 
-    return _response
+@pytest.fixture
+def cloudera_version(request):
+    return ApiVersionInfo() if request.param is None else ApiVersionInfo(version=request.param)
 
 
-def get_timeseries_resource():
-    return [
-        ApiTimeSeriesResponseList(
-            items=[
-                ApiTimeSeriesResponse(
-                    time_series=[
-                        ApiTimeSeries(
-                            data=[
-                                ApiTimeSeriesData(value=49.7),
-                            ],
-                            metadata=ApiTimeSeriesMetadata(attributes={'category': category}, alias=metric),
-                        )
-                        for metric in metrics
-                    ]
-                ),
-            ],
-        )
-        for category, metrics in TIMESERIES_METRICS.items()
-    ]
+@pytest.fixture
+def cloudera_version_5_0_0():
```

Comment:
I noticed that this fixture is not actually being used. Should we just remove it?

Reply:
You are right. I will remove it.